src/v/serde/parquet/schema.h

```diff
@@ -261,7 +261,7 @@ struct schema_element {
      *
      * This must be set to required on the schema root
      */
-    field_repetition_type repetition_type;
+    field_repetition_type repetition_type{field_repetition_type::required};
```

Comment:
This should not have a default - it should be required to be always set explicitly. Where are we forgetting to set this value in schema translation?

Reply:
Okay I could be convinced this is the right default, but I really think callers should be explicitly setting this.

Reply:
Yea agreed about callers needing to think about this. Maybe we should restrict how schemas are constructed to force that? As is, we're pretty loose about it. I'll leave it for now though since the bad behavior now is probably better than UB

I did track the issue to how we were handling structs

Reply:
Awesome - well agreed bad behavior is better than UB. I will think about better APIs for constructing schemas...